rudder-web/src/main/webapp/templates-hidden/common-layout.html

```diff
@@ -131,124 +131,127 @@
                         <div class="filter-search col-xs-12">
                           <div class="panel panel-default panel-help">
                             <div class="panel-heading heading-search">
-                              <h4 class="panel-title">Type</h4>
-                                <div class="btn-toolbar" role="toolbar">
+                              <div class="btn-toolbar" role="toolbar">
                                 <div class="btn-group group-all" data-toggle="buttons">
                                   <label for="filter-all" class="btn btn-default" ng-click="$parent.checkFilter($event, true,$parent.filter.all.activated,'all');" ng-class="{active : $parent.filter.all.activated || (!$parent.filter.directive.activated && !$parent.filter.group.activated && !$parent.filter.node.activated && !$parent.filter.parameter.activated && !$parent.filter.rule.activated)};">
                                     <input type="radio" id="filter-all" checked="true" ng-checked="$parent.filter.all.activated || (!$parent.filter.directive.activated && !$parent.filter.group.activated && !$parent.filter.node.activated && !$parent.filter.parameter.activated && !$parent.filter.rule.activated);"/>All
-                                    <span class="badge pull-right" ng-class="{'label-success':$parent.filter.all.nbResults>0}">{{$parent.filter.all.nbResults}}</span>
+                                    <span class="badge pull-right">
+                                      <span ng-show="((!searching) || (searching && 2 >= searchStr.length)) && ($parent.filter.all.activated || (!$parent.filter.directive.activated && !$parent.filter.group.activated && !$parent.filter.node.activated && !$parent.filter.parameter.activated && !$parent.filter.rule.activated));">
```

Comment:
When conditions are that complicated, you can put them in the angular file as a dunction and call it. (maybe it could allow to refactor  all the complex conditions ... like '(!$parent.filter.directive.activated && !$parent.filter.group.activated && !$parent.filter.node.activated && !$parent.filter.parameter.activated && !$parent.filter.rule.activated)' in a function 'checkParentfilter"